src/rules/no-native-attributes.ts

```diff
@@ -20,103 +56,29 @@ const rule: Rule.RuleModule = {
     }
   },
 
   create(context): Rule.RuleListener {
     // variables should be defined here
 
-    // Taken from https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes
-    const NATIVE_ATTRS = [
-      'accesskey',
-      'autocapitalize',
-      'autofocus',
-      'class',
-      'contenteditable',
-      'contextmenu',
-      'dir',
-      'draggable',
-      'enterkeyhint',
-      'exportparts',
-      'hidden',
-      'id',
-      'inert',
-      'inputmode',
-      'is',
-      'itemid',
-      'itemprop',
-      'itemref',
-      'itemscope',
-      'itemtype',
-      'lang',
-      'nonce',
-      'part',
-      'role',
-      'slot',
-      'spellcheck',
-      'style',
-      'tabindex',
-      'title',
-      'translate',
-      'virtualkeyboardpolicy'
-    ];
-
     //----------------------------------------------------------------------
     // Public
     //----------------------------------------------------------------------
 
     return {
       'ClassExpression,ClassDeclaration': (node: ESTree.Class): void => {
-        for (const member of node.body.body) {
-          if (
-            member.type === 'MethodDefinition' &&
-            member.kind === 'get' &&
-            member.static &&
-            member.key.type === 'Identifier' &&
-            member.key.name === 'properties'
-          ) {
-            const returnValue = member.value.body.body
-              .find((statement) => statement.type === 'ReturnStatement');
-
-            if (returnValue) {
-              const object = returnValue.argument;
-
-              if (object) {
-                for (const property of object.properties) {
-                  if (
-                    property.type === 'Property' &&
-                    NATIVE_ATTRS.includes((property.key as ESTree.Identifier).name)
-                  ) {
-                    context.report({
-                      node: member,
-                      messageId: 'noNativeAttributes'
-                    });
-                  }
-                }
-              }
-            }
-          }
+        const propertyMap = getPropertyMap(node);
 
-          if (
-            member.type === 'PropertyDefinition' &&
-            member.static &&
-            member.key.type === 'Identifier' &&
-            member.key.name === 'properties'
-          ) {
-            const value = member.value as ESTree.ObjectExpression;
-            for (const property of value.properties) {
-              if (
-                property.type === 'Property' &&
-                NATIVE_ATTRS.includes((property.key as ESTree.Identifier).name)
-              ) {
-                context.report({
-                  node: member,
-                  messageId: 'noNativeAttributes'
-                });
-              }
-            }
+        for (const [prop] of propertyMap.entries()) {
+          if (NATIVE_ATTRS.includes(prop)) {
+            context.report({
+              node: node,
```

Comment:
have opened #161 for this FYI so we can report the exact key node rather than the whole class here

Reply:
nice 👌